### PR TITLE
Refactored existing Questionnaire linked test to cover new functionality

### PR DESCRIPTION
**Pull request by AdeHarrison**
# How to test?
as part of [pr](https://github.com/ONSdigital/census-rm-case-processor/pull/41)

# Links
[Trello](https://trello.com/c/HaHa9jE7/826-rmc-142-implement-questionnaire-linked-event-13)

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by AdeHarrison** [comment]
[RMC-142 Implement Questionnaire Linked Event (13)](https://trello.com/c/HaHa9jE7/826-rmc-142-implement-questionnaire-linked-event-13)

File: acceptance_tests/features/steps/unaddressed.py [hunk 1]
```diff
@@ -63,6 +63,21 @@ def check_uac_message_is_received(context):
...
 
+@then("a CaseUpdated message is emitted to RH and Action Scheduler")
+def check_case_updated_message_is_emitted(context):
+    time.sleep(2)  # Give case processor a chance to process the receipt event
+
+    # skip receipt case updated message
+    start_listening_to_rabbit_queue(Config.RABBITMQ_RH_OUTBOUND_CASE_QUEUE_TEST,
+                                    functools.partial(_questionnaire_linked_callback, context=context))
+
+    start_listening_to_rabbit_queue(Config.RABBITMQ_RH_OUTBOUND_CASE_QUEUE_TEST,
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/unaddressed.py:72]: why does this get called a second time?

File: acceptance_tests/features/unaddressed.feature [hunk 2]
```diff
@@ -4,11 +4,13 @@ Feature: Generating UAC/QID pairs for unaddressed letters & questionnaires
...
 
-  Scenario: Questionnaire linked to unaddressed
+  Scenario: Questionnaire linked to case
     Given sample file "sample_for_questionnaire_linked.csv" is loaded
     And messages are emitted to RH and Action Scheduler with [01] qids
-    When a Questionnaire Linked message is sent
+    When the receipt msg for a created case is put on the GCP pubsub
+    And a Questionnaire Linked message is sent
     Then a Questionnaire Linked event is logged
+    And a CaseUpdated message is emitted to RH and Action Scheduler
```
Comment (AdamHawtin) [inline acceptance_tests/features/unaddressed.feature:13]: Test should check for the UAC Updated message with the new link being sent.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]

File: acceptance_tests/features/steps/unaddressed.py (hunk 1 above)
Comment (AdeHarrison) [reply acceptance_tests/features/steps/unaddressed.py:72]: sorted

File: acceptance_tests/features/unaddressed.feature (hunk 2 above)
Comment (AdeHarrison) [reply acceptance_tests/features/unaddressed.feature:13]: done

File: acceptance_tests/features/steps/unaddressed.py [hunk 3]
```diff
@@ -29,7 +29,10 @@ def send_unaddressed_message(context, questionnaire_type):
 
 @when("a Questionnaire Linked message is sent")
 def check_linked_message_is_received(context):
-    context.linked_case = context.case_created_events[1]['payload']['collectionCase']
```
Comment (rjweeks) [inline acceptance_tests/features/steps/unaddressed.py:32]: This is sending a message but not checking a linked message is received? Function name should be changed.

File: acceptance_tests/features/unaddressed.feature (hunk 2 above)
Comment (AdamHawtin) [reply acceptance_tests/features/unaddressed.feature:13]: I can't see it checking for the UAC Updated event anywhere? I definitely think that should be tested for, it's an critical part of this feature that RH etc are notified of the link.

**Pull request closed without merging**